crates/iota-sdk-ffi/src/crypto/secp256r1.rs

```diff
@@ -0,0 +1,161 @@
+    #[uniffi::constructor]
+    pub fn new(bytes: Vec<u8>) -> Result<Self> {
+        Ok(Self(iota_crypto::secp256r1::Secp256r1PrivateKey::new(
+            bytes.try_into().map_err(|v: Vec<u8>| {
+                SdkFfiError::custom(format!("expected bytes of length 32, found {}", v.len()))
+            })?,
+        )))
+    }
+
+    pub fn scheme(&self) -> SignatureScheme {
+        self.0.scheme()
+    }
+
+    /// Get the public key corresponding to this private key.
+    pub fn public_key(&self) -> Secp256r1PublicKey {
+        self.0.public_key().into()
+    }
+
+    /// Sign a message and return a Secp256r1Signature.
+    pub fn try_sign(&self, message: &[u8]) -> Result<Secp256r1Signature> {
+        Ok(
+            iota_crypto::Signer::<iota_types::Secp256r1Signature>::try_sign(&self.0, message)?
+                .into(),
+        )
+    }
+
+    /// Sign a message and return a SimpleSignature.
+    pub fn try_sign_simple(&self, message: &[u8]) -> Result<SimpleSignature> {
+        Ok(iota_crypto::Signer::<iota_types::SimpleSignature>::try_sign(&self.0, message)?.into())
+    }
+
+    /// Sign a message and return a UserSignature.
+    pub fn try_sign_user(&self, message: &[u8]) -> Result<UserSignature> {
+        Ok(iota_crypto::Signer::<iota_types::UserSignature>::try_sign(&self.0, message)?.into())
+    }
+
+    pub fn verifying_key(&self) -> Secp256r1VerifyingKey {
+        self.0.verifying_key().into()
+    }
+
+    /// Generate a new random Secp256r1PrivateKey
+    #[uniffi::constructor]
+    pub fn generate() -> Self {
+        Self(iota_crypto::secp256r1::Secp256r1PrivateKey::generate(OsRng))
+    }
+
+    /// Deserialize PKCS#8 private key from ASN.1 DER-encoded data (binary
+    /// format).
+    #[uniffi::constructor]
+    pub fn from_der(bytes: Vec<u8>) -> Result<Self> {
+        Ok(iota_crypto::secp256r1::Secp256r1PrivateKey::from_der(&bytes)?.into())
+    }
+
+    /// Serialize this private key as DER-encoded PKCS#8
+    pub fn to_der(&self) -> Result<Vec<u8>> {
+        Ok(self.0.to_der()?)
+    }
+
+    /// Deserialize PKCS#8-encoded private key from PEM.
+    #[uniffi::constructor]
+    pub fn from_pem(s: &str) -> Result<Self> {
+        Ok(iota_crypto::secp256r1::Secp256r1PrivateKey::from_pem(s)?.into())
+    }
+
+    /// Serialize this private key as PEM-encoded PKCS#8
+    pub fn to_pem(&self) -> Result<String> {
+        Ok(self.0.to_pem()?)
+    }
+}
+
+#[derive(derive_more::From, uniffi::Object)]
+pub struct Secp256r1VerifyingKey(pub iota_crypto::secp256r1::Secp256r1VerifyingKey);
+
+#[uniffi::export]
+impl Secp256r1VerifyingKey {
+    #[uniffi::constructor]
+    pub fn new(public_key: &Secp256r1PublicKey) -> Result<Self> {
+        Ok(iota_crypto::secp256r1::Secp256r1VerifyingKey::new(public_key)?.into())
+    }
+
+    pub fn public_key(&self) -> Secp256r1PublicKey {
+        self.0.public_key().into()
+    }
+
+    pub fn verify(&self, message: Vec<u8>, signature: &Secp256r1Signature) -> bool {
```

Comment:
Aren't we missing two verify methods?

Reply:
https://github.com/iotaledger/iota-rust-sdk/pull/126/commits/bf7d39b0acf86e99de20462355b18e809de67b1c